src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -1518,58 +1482,50 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     }
   }
 
-  function getPriorityContext(
+  function getExpirationTime(
     fiber: Fiber,
     forceAsync: boolean,
-  ): PriorityLevel | null {
-    if (isPerformingWork && !isCommitting) {
-      // Updates during the render phase should expire at the same time as
-      // the work that is being rendered. Return null to indicate.
-      return null;
-    }
-
-    let priorityLevel = priorityContext;
-    if (priorityLevel === NoWork) {
+  ): ExpirationTime {
+    let expirationTime;
+    if (expirationContext !== NoWork) {
+      // An explicit expiration context was set;
+      expirationTime = expirationContext;
+    } else if (isPerformingWork) {
+      if (isCommitting) {
+        // Updates that occur during the commit phase should have task priority
+        // by default.
+        expirationTime = Task;
+      } else {
+        // Updates during the render phase should expire at the same time as
+        // the work that is being rendered.
+        expirationTime = nextRenderExpirationTime;
```

Comment:
From the comment, I'm a little confused why we use `else` rather than `else if (isRendering)`, @acdlite would you mind explain some details? Thanks!